frontend/src/components/@shared/Time.tsx

```diff
@@ -1,32 +1,122 @@
-const timesByThirty: string[] = [];
+import styled from '@emotion/styled';
+import { useState } from 'react';
 
-for (let i = 1; i < 25; i += 1) {
-  for (let j = 0; j < 2; j += 1) {
-    timesByThirty.push(`${String(i).padStart(2, '0')}:${String(j * 30).padStart(2, '0')}`);
+type TimeTable = {
+  time: string;
+  isPassed: boolean;
+};
+
+const timeTableGenerator = (startHour, endHour) => {
+  const timeTable: TimeTable[] = [];
+
+  for (let i = startHour; i < endHour + 1; i += 1) {
+    for (let j = 0; j < 2; j += 1) {
+      if (i === endHour) {
+        break;
+      }
+      const time = `${String(i).padStart(2, '0')}:${String(j * 30).padStart(2, '0')}`;
```

Comment:
이거 시간단위가 30분이 아니라 10분이나 다른 단위로 바뀌면 좀 힘들지 않을까?
```suggestion
    const scale = 2;
    for (let j = 0; j < scale; j += 1) {
      if (i === endHour) {
        break;
      }
      const time = `${String(i).padStart(2, '0')}:${String(j * 60/scale).padStart(2, '0')}`;
```

간단하게 이런식으로 해도 좋을듯.
2중 for문은 없앨 수 있을 거 같은데,, 당장은 생각이 안나네

Reply:
적용해봤습니다 굿굿